react/src/components/QrReaderOverlay/OrReaderOverlay.test.tsx

```diff
@@ -0,0 +1,332 @@
+import { GraphQLError } from "graphql";
+import "@testing-library/jest-dom";
+import userEvent from "@testing-library/user-event";
+import { screen, render } from "tests/test-utils";
+import HeaderMenuContainer from "components/HeaderMenu/HeaderMenuContainer";
+import { useAuth0 } from "@auth0/auth0-react";
+import { QrReader } from "components/QrReader/QrReader";
+import { mockMatchMediaQuery } from "mocks/functions";
+import { mockAuthenticatedUser } from "mocks/hooks";
+import { mockImplementationOfQrReader } from "mocks/components";
+import {
+  BOX_DETAILS_BY_LABEL_IDENTIFIER_QUERY,
+  GET_BOX_LABEL_IDENTIFIER_BY_QR_CODE,
+} from "queries/queries";
+import { generateMockBox } from "mocks/boxes";
+
+jest.mock("@auth0/auth0-react");
+jest.mock("components/QrReader/QrReader");
+
+// .mocked() is a nice helper function from jest for typescript support
+// https://jestjs.io/docs/mock-function-api/#typescript-usage
+const mockedUseAuth0 = jest.mocked(useAuth0);
+const mockedQrReader = jest.mocked(QrReader);
+
+beforeEach(() => {
+  // setting the screensize to
+  mockMatchMediaQuery(true);
+  mockAuthenticatedUser(mockedUseAuth0, "dev_volunteer@boxaid.org");
+});
+
+const queryFindNoBoxAssociated = {
+  request: {
+    query: BOX_DETAILS_BY_LABEL_IDENTIFIER_QUERY,
+    variables: {
+      labelIdentifier: "123456",
+    },
+  },
+  result: {
+    data: {
+      box: null,
+    },
+    errors: [new GraphQLError("Error!", { extensions: { code: "BAD_USER_INPUT" } })],
+  },
+};
+
+it("3.4.1.2 - Mobile: Enter invalid box identifier and click on Find button", async () => {
+  const user = userEvent.setup();
+  mockImplementationOfQrReader(mockedQrReader, "NoBoxAssociatedWithQrCode");
+  // mock scanning a QR code
+  render(<HeaderMenuContainer />, {
+    routePath: "/bases/:baseId",
+    initialUrl: "/bases/1",
+    mocks: [queryFindNoBoxAssociated],
+    additionalRoute: "/bases/1/boxes/123456",
+  });
+
+  // Open QROverlay
```

Comment:
Can you put a comment that this is actually test case 3.4.1.1?

Reply:
done